### PR TITLE
Add -R option

### PR DESCRIPTION
**Pull request by papachristoumarios**
The -R option can produce the needed call graphs and then exit. 

Example

```
cscout -R cgraph.txt -R fgraph.txt\?gtype\=C awk.cs
```

This PR includes
 - Implementation for the -R option
 - GraphDisplay can have arguments as fields
 - `frgaph_page`, `cgraph_page` can work in a more generic way if swill is not used
 - Manpages updated

### CODE REVIEW

File: src/cscout.cpp [hunk 1]
```diff
@@ -2372,18 +2395,62 @@ vector<string> split_by_delimiter(string &s, char delim) {
...
+			continue;
+		}
+		FILE *target = fopen(split_base_and_opts[0].c_str() , "w+");
+		string base = split_base_and_opts[0];
+		GDTxt gd(target);
+		// Disable swill
+		gd.uses_swill = false;
+		vector<string> opts;
+
+		if (split_base_and_opts.size() == 1) goto call_graph_functions;
```
Comment (dspinellis) [inline src/cscout.cpp]: Put the code below in a separate function  to avoid the goto.

File: src/cscout.cpp [hunk 2]
```diff
@@ -3348,6 +3366,9 @@ main(int argc, char *argv[])
 	}
 
 	if (process_mode != pm_compile) {
+
```
Comment (dspinellis) [inline src/cscout.cpp]: Please eliminate the blank lines

File: src/cscout.cpp [hunk 3]
```diff
@@ -3434,6 +3436,16 @@ main(int argc, char *argv[])
 	}
 #endif
 
+	if (process_mode == pm_r_option) {
+		cerr << "Producing call graphs" << endl;
+
+		for (const char *d : call_graphs) {
+			cerr << d << endl;
```
Comment (dspinellis) [inline src/cscout.cpp]: Shouldn't these go to the specified files? Otherwise, shouldn't it be `cout`.

File: src/cscout.cpp [hunk 4]
```diff
@@ -2358,10 +2358,33 @@ graph_pdf_page(FILE *fo, void (*graph_fun)(GraphDisplay *))
...
 }
 
+
+vector<string> split_by_delimiter(string &s, char delim) {
+	string buf;                 // Have a buffer string
+	stringstream ss(s);       // Insert the string into a stream
+
+	vector<string> tokens; // Create vector to hold our words
+
+	while(getline(ss, buf, delim))
```
Comment (dspinellis) [inline src/cscout.cpp]: Space after `while`

File: src/cscout.cpp [hunk 5]
```diff
@@ -2358,10 +2358,33 @@ graph_pdf_page(FILE *fo, void (*graph_fun)(GraphDisplay *))
...
+
 // Produce call graphs with -R option
 static void produce_call_graphs(const vector <string> &call_graphs)
 {
+	char base_splitter = '?';
 
+	for (string url: call_graphs) {
+		vector<string> split_base_and_opts = split_by_delimiter(url, base_splitter);
+		for (string z: split_base_and_opts) {
+			cerr << z << endl;
```
Comment (dspinellis) [inline src/cscout.cpp]: Output should go to a file, right?

File: src/cscout.cpp (hunk 3 above)
Comment (papachristoumarios) [reply src/cscout.cpp]: This is for logging purposes so according to the philosophy it ought to go to stderr.

Comment (dspinellis) [reply src/cscout.cpp]: OK, ignore.  I though it was the actual graph.

File: src/cscout.cpp (hunk 1 above)
Comment (papachristoumarios) [reply src/cscout.cpp]: It is now in an if-statement

File: src/cscout.cpp (hunk 5 above)
Comment (papachristoumarios) [reply src/cscout.cpp]: This is an old commit. I was testing functionality. It is not included in the final version.

File: src/cscout.cpp (hunk 2 above)
Comment (papachristoumarios) [reply src/cscout.cpp]: Done!

**Comment by papachristoumarios** [comment]
@dspinellis  Changes have been issued.

**Pull request merged**